Lib/shelve.py

```diff
@@ -91,6 +96,15 @@ def __init__(self, dict, protocol=None, writeback=False,
         self.cache = {}
         self.keyencoding = keyencoding
 
+        if serializer is None and deserializer is None:
+            self.serializer = dumps
+            self.deserializer = loads
+        elif (serializer is None and deserializer is not None) or (deserializer is None and serializer is not None):
```

Comment:
This with be clearer with XOR:

`elif (serializer is None) ^ (deserializer is None):`